### DISCUSSION TITLE
build maven:unable to find any built artifacts for pattern(s):

### DISCUSSION THREAD

**Discussion by AmbitiousHERO** [paketo-buildpacks/feedback#198] [Java Team]
unable to invoke layer creator
unable to contribute application layer
unable to resolve artifacts
unable to find any built artifacts for pattern(s):
target/*.[ejw]ar
ERROR: failed to build: exit status 1
ERROR: failed to build: executing lifecycle: failed with status code: 51

**Comment by dmikusa** [comment]
Can you post more information? The full build output and command you're running would be helpful.

All I can say here is that your build ran, and the buildpack could not find either `target/*.war` or `target/*.jar` to execute.

**Reply by AmbitiousHERO** [reply]
command:  
pack build --path ./srcs/sona-gateway/  avicit/sona-gateway:1.0-SNAPSHOT --env BP_JVM_VERSION=11 \
    --env "BP_MAVEN_BUILD_ARGUMENTS=clean package -P buildImage "  \
    --volume /home/sona/sona-deploy2/build/dependencies/cnb/offline-packages:/platform/bindings/dependency-mapping \
    --volume /home/sona/sona-deploy2/build/dependencies/m2:/home/cnb/.m2:rw \
    --builder paketobuildpacks/builder:base --pull-policy never

results:
[INFO] --- compiler:3.8.1:compile (default-compile) @ sona-gateway-starter ---
      [INFO] Changes detected - recompiling the module!
      [INFO] Compiling 20 source files to /workspace/sona-gateway-starter/target/classes
      [INFO] /workspace/sona-gateway-starter/src/main/java/com/avic/sona/auth/AuthConfiguration.java: /workspace/sona-gateway-starter/src/main/java/com/avic/sona/auth/AuthConfiguration.java uses unchecked or unsafe operations.
      [INFO] /workspace/sona-gateway-starter/src/main/java/com/avic/sona/auth/AuthConfiguration.java: Recompile with -Xlint:unchecked for details.
      [INFO] 
      [INFO] --- resources:3.3.0:testResources (default-testResources) @ sona-gateway-starter ---
      [INFO] skip non existing resourceDirectory /workspace/sona-gateway-starter/src/test/resources
      [INFO] 
      [INFO] --- compiler:3.8.1:testCompile (default-testCompile) @ sona-gateway-starter ---
      [INFO] No sources to compile
      [INFO] 
      [INFO] --- surefire:2.19.1:test (default-test) @ sona-gateway-starter ---
      [INFO] Tests are skipped.
      [INFO] 
      [INFO] --- jar:3.3.0:jar (default-jar) @ sona-gateway-starter ---
      [INFO] Building jar: /workspace/sona-gateway-starter/target/sona-gateway-starter-1.230626-SNAPSHOT.jar
      [INFO] ------------------------------------------------------------------------
      [INFO] Reactor Summary for sona-gateway 1.230626-SNAPSHOT:
      [INFO] 
      [INFO] sona-gateway ....................................... SUCCESS [  0.099 s]
      [INFO] sona-gateway-starter ............................... SUCCESS [  4.208 s]
      [INFO] ------------------------------------------------------------------------
      [INFO] BUILD SUCCESS
      [INFO] ------------------------------------------------------------------------
      [INFO] Total time:  4.585 s
      [INFO] Finished at: 2023-07-07T03:04:38Z
      [INFO] ------------------------------------------------------------------------
      [WARNING] The requested profile "buildImage" could not be activated because it does not exist.
      
unable to invoke layer creator
unable to contribute application layer
unable to resolve artifacts
unable to find any built artifacts for pattern(s):
target/*.[ejw]ar
ERROR: failed to build: exit status 1
ERROR: failed to build: executing lifecycle: failed with status code: 51


But I can use local Maven build success

**Reply by dmikusa** [reply]
It looks like you have a multi-module project. You need to set `BP_MAVEN_BUILT_ARTIFACT`. Try adding `-e "BP_MAVEN_BUILT_ARTIFACT=sona-gateway-starter/target/sona-gateway-starter-*.jar"`. That will change where the buildpack looks for the JAR file.

**Reply by AmbitiousHERO** [reply]
I successfully completed the build，Thank you very much for your help!
